site/src/util/zipGeocoding.ts

```diff
@@ -0,0 +1,46 @@
+import { StateCode, USZipCode } from "../components/address-input/SmartyStreets";
+
+const smartyApiURL = "https://us-zipcode.api.smartystreets.com/lookup";
+const smartyAuthId = (process.env.SMARTY_STREETS_KEY as string);
+
+export default async (zipcode: string) => {
+   const url = new URL(smartyApiURL);
+   url.searchParams.append("auth-id", smartyAuthId);
+   url.searchParams.append("zipcode", zipcode);
+   return fetch(url.toString()).
+      then(
+         response => response.json(),
+         error => console.log("SmartyStreets Error:", error),
+      ).then((response: USZipCode.QueryResult) => {
+         let result: USZipCode.QueryResultItem = response[0];
+
+         if(result.reason) { return { error: result.reason }; }
+
+         let cities = new Set<string>();
+         let counties = new Set<string>();
+         let states = new Map<StateCode, string>(); // Abbreviation to full name
+
+         let defaultCityState = (result.zipcodes as USZipCode.ZipCode[])[0];
+
+         if(result.city_states) {
+            result.city_states.forEach((city: USZipCode.CityState) => cities.add(city.city));
+         }
+
+         counties.add(defaultCityState.county_name);
+         states.set(defaultCityState.state_abbreviation, defaultCityState.state);
+
+         if (defaultCityState.alternate_counties) {
+            defaultCityState.alternate_counties.forEach((county: USZipCode.AlternateCounty) => {
+               counties.add(county.county_name);
+               states.set(county.state_abbreviation, county.state);
+            });
+         }
+
+         return {
+            zip: defaultCityState.zipcode,
+            cities: cities,
+            states: states,
+            counties: counties,
+         };
+   });
+};
```

Comment:
Can just move this file to the address-input component since that's the only place its used

Reply:
Think it makes sense here to separate an API service from presentational concerns, but I can move it back if you feel strongly.

Reply:
If you want to make a SmartyStreets namespace or class or module beneath `/address-input` instead of the member functions on the component that would be good :+1: 

But components aren't just presentation, they compose presentation and logic, so since the SmartyStreets stuff is entirely contained to the `address-input` component we can keep it all there. E.g., we could pull it out into its own component just like we did the form itself.

Reply:
Moved `zipGeocoding` into `src/components/address-input`